server/src/main/java/org/elasticsearch/rest/action/document/RestBulkAction.java

```diff
@@ -107,6 +276,6 @@ public boolean supportsBulkContent() {
 
     @Override
     public boolean allowsUnsafeBuffers() {
-        return true;
+        return false;
```

Comment:
This should be true. I noticed stream didnt handle 400 bad request properly, fixed here https://github.com/elastic/elasticsearch/pull/112227